### PR TITLE
Create single module, aggregate dependencies

### LINKED ISSUE #263: Create single module, aggregate dependencies
Currently, we have many modules, for which we have `go.mod`s, where the dependencies for each module are kept. This make upgrading some dependencies such as firecracker-containerd error prone, because the dependency needs to be changed in several places.

It seems a lot of the dependencies overlap across the modules - containerd, firecracker-containerd, and some less obvious ones, like grpc. This is one reason why I propose to keep the non-top modules as packages - `misc`, `taps`, etc. Another reason is that really these modules are not meant to be used outside of vHive in order to warrant keeping them as modules if someone wants to use them.

I propose the following way to do this:
1. Delete `go.mod` and `go.sum` in all modules we want to keep as packages. Import paths should stay the same.
2. Introduce the missing dependencies with version to the top-level `go.mod`.

### PR DESCRIPTION
**Pull request by boramalper**
Fixes #263.

I'll rebase them into a single commit, left as-is for clarity for now.

### CODE REVIEW

File: ctriface/iface_test.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ import (
 var (
 	isUPFEnabled = flag.Bool("upf", false, "Set UPF enabled")
 	isLazyMode   = flag.Bool("lazy", false, "Set lazy serving on or off")
-	isWithCache  = flag.Bool("withCache", false, "Do not drop the cache before measurements")
+	// isWithCache  = flag.Bool("withCache", false, "Do not drop the cache before measurements")
```
Comment (ustiugov) [inline ctriface/iface_test.go:43]: we need this option

File: memory/manager/manager_test.go [hunk 2]
```diff
@@ -20,6 +20,7 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
+//nolint:unused
```
Comment (ustiugov) [inline memory/manager/manager_test.go:23]: I agree with this workaround, however please raise an Issue

Comment (boramalper) [reply memory/manager/manager_test.go]: Done, see #270.

**Review by ustiugov — CHANGES_REQUESTED** [review]
Please make the commits' comments slightly more verbose. it is not clear what you do with each package from commit messages.

Rest is good, thanks.

**Comment by ustiugov** [comment]
please rebase

**Comment by boramalper** [comment]
@ustiugov 2 tests are failing but otherwise lgtm, care to merge?

**Review by ustiugov — CHANGES_REQUESTED** [review]
More minor things:

- [x] add a not to Changed in the changelog (simplified go dependencies management in a single file)
- [x] remove the corresponding module from dependabot's config: https://github.com/ease-lab/vhive/blob/main/.github/dependabot.yml

**Review by ustiugov — APPROVED** [review]

**Pull request merged**